chromium_src/chrome/browser/extensions/extension_management.cc

```diff
@@ -3,10 +3,48 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// Note: BuildServiceInstanceFor is an overriden virtual method, so we need to
```

Comment:
This is something we generally try to avoid and I think in this case it's probably not worth the potential risk of a runtime static cast issue.

Reply:
> This is something we generally try to avoid and I think in this case it's probably not worth the potential risk of a runtime static cast issue.

Sounds good, thanks. Will close the issue and try to avoid it in the future